torchgeo/datamodules/utils.py

```diff
@@ -3,13 +3,17 @@
 
 """Common datamodule utilities."""
 
-from typing import Any, List, Optional
+from typing import Any, List, Optional, Union
 
-from torch.utils.data import Dataset, Subset, random_split
+from torch.utils.data import Subset, TensorDataset, random_split
+
+from ..datasets import VisionDataset
 
 
 def dataset_split(
-    dataset: Dataset[Any], val_pct: float, test_pct: Optional[float] = None
+    dataset: Union[TensorDataset, VisionDataset],
```

Comment:
So `Dataset` doesn't have a `__len__` method, so this wasn't valid. Also, `GeoDataset` isn't indexed by integer, so I don't think that works either. I wish there was a way to say "any `Dataset` with a `__len__` method and integer indexing" but I don't know a better way to do this.